src/modules/product/interface/product.routes.ts

```
@@ -22,60 +23,110 @@ router.get('/', (req, res, next) => productController.getProducts(req, res, next
+);
 
 // Protected routes
 router.use(authenticate);
 
 // Helper function to handle file upload and product data
-const handleProductUpload = async (req: Request, res: Response, next: NextFunction, controllerMethod: Function) => {
+const handleProductUpload = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+  controllerMethod: Function
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)